### PR TITLE
badgeV1

### PR DESCRIPTION
**Pull request by lukeac123**
- rename "BadgeContent" to "value"
- inline styling when no child is passed to the component
- truncation on numbers > 4 characters or when max value is exceeded
- truncation on strings > 4 characters

### CODE REVIEW

**Comment by lukeac123** [comment]
- Waiting for final sign off from design
- once approved will add a changeset and merge to core

File: packages/lab/src/badge/Badge.css
Comment (libertymayc) [inline packages/lab/src/badge/Badge.css:24]:
```suggestion
```
defined globally so we don't need it here

File: packages/lab/src/badge/Badge.css
Comment (libertymayc) [inline packages/lab/src/badge/Badge.css:32]: these are quite specific, are they prone to breaking?

File: packages/lab/src/badge/Badge.css
Comment (ZarkoPernar) [reply packages/lab/src/badge/Badge.css:24]: Didn't we have an issue with a project overwriting font styles with an html selector which trumps global @libertymayc ?

File: packages/lab/stories/badge/badge.stories.tsx
Comment (ZarkoPernar) [inline packages/lab/stories/badge/badge.stories.tsx:76]:
```suggestion
TruncatedString.args = {
```

File: packages/lab/stories/badge/badge.stories.tsx
Comment (ZarkoPernar) [inline packages/lab/stories/badge/badge.stories.tsx:36]: Does it look the same without this?

File: packages/lab/src/badge/Badge.css
Comment (ZarkoPernar) [inline packages/lab/src/badge/Badge.css:14]: As per Navs comment
```suggestion
  min-width: var(--salt-accent-lineHeight);
```

Comment (libertymayc) [reply packages/lab/src/badge/Badge.css:14]: line height on minwidth? 😛

Comment (joshwooding) [reply packages/lab/src/badge/Badge.css:14]: This is fine

File: packages/lab/stories/badge/badge.stories.tsx
Comment (ZarkoPernar) [reply packages/lab/stories/badge/badge.stories.tsx:36]: Nope, you are right. All good 👍

File: packages/lab/src/badge/Badge.css
Comment (libertymayc) [reply packages/lab/src/badge/Badge.css:24]: did we?

**Review by lilyvc — APPROVED** [review]

File: packages/lab/src/badge/Badge.css
Comment (lukeac123) [reply packages/lab/src/badge/Badge.css:32]: seems to be fine on all the stories but how else would you suggest doing the positioning ?

**Pull request closed without merging**
